### PR TITLE
234396 step 2 validation errors

### PR DESCRIPTION
**Pull request by JasmeetKaur-Defra**

### CODE REVIEW

File: EST.MIT.Web/Pages/invoice/AddInvoiceLine/AddInvoiceLine.razor [hunk 1]
```diff
@@ -4,6 +4,22 @@
 @using EST.MIT.Web.Shared.Components.GDSInputText
 @using Helpers
 
+@functions {
+    string HasErrorMessage(string key)
```
Comment (andygibbonsdefra) [inline EST.MIT.Web/Pages/invoice/AddInvoiceLine/AddInvoiceLine.razor:8]: Name "HasErrorMessages" implies a boolean return. Should be maybe "ErrorMessagesForField(string fieldKey) or something

File: EST.MIT.Web/Pages/invoice/AddInvoiceLine/AddInvoiceLine.razor [hunk 2]
```diff
@@ -4,6 +4,22 @@
 @using EST.MIT.Web.Shared.Components.GDSInputText
 @using Helpers
 
+@functions {
```
[the remark this answers is not in the document]
Comment (JasmeetKaur-Defra) [reply EST.MIT.Web/Pages/invoice/AddInvoiceLine/AddInvoiceLine.razor]: Moved the function to cs

File: EST.MIT.Web/Pages/invoice/AddInvoiceLine/AddInvoiceLine.razor (hunk 1 above)
Comment (JasmeetKaur-Defra) [reply EST.MIT.Web/Pages/invoice/AddInvoiceLine/AddInvoiceLine.razor]: Renamed the function name

File: EST.MIT.Web/Pages/invoice/AddPaymentRequest/AddPaymentRequest.razor.cs [hunk 3]
```diff
@@ -51,7 +51,7 @@ private async Task SaveHeader()
         }
```
Comment (andygibbonsdefra) [inline EST.MIT.Web/Pages/invoice/AddPaymentRequest/AddPaymentRequest.razor.cs:52]: Think this 'setting errors' should be in an Else block, as I think the code falls through to it if response.IsSuccess. (try it). Also might need to set IsErrored=false and clear errors in the success block...?

Comment (JasmeetKaur-Defra) [reply EST.MIT.Web/Pages/invoice/AddPaymentRequest/AddPaymentRequest.razor.cs]: That's done for both payment request and invoice line

File: EST.MIT.Web/Pages/invoice/AddPaymentRequest/AddPaymentRequest.razor [hunk 4]
```diff
@@ -6,25 +6,39 @@
...
+    <InvoiceMetaData Invoice=invoice />
 
     <section class="govuk-width-container">
-        
-        @if(IsErrored)
+
+        @if (errors.Count > 0)
         {
-            <div class="govuk-form-group--error">
-                @foreach (var error in errors.Values.SelectMany(x => x).Distinct())
```
Comment (andygibbonsdefra) [inline EST.MIT.Web/Pages/invoice/AddPaymentRequest/AddPaymentRequest.razor:24]: Might need this Distinct() sort of functionality. Omitting any FRN/SBI/Vendor customer key and saving shows duplicate messages.

Comment (JasmeetKaur-Defra) [reply EST.MIT.Web/Pages/invoice/AddPaymentRequest/AddPaymentRequest.razor:24]: That's fixed

**Review by andygibbonsdefra — APPROVED** [review]

**Pull request merged**